src/main/java/opendcs/dao/ScheduleEntryDAO.java

```diff
@@ -56,489 +56,596 @@
-				+ "(select schedule_entry_id from schedule_entry where loading_application_id = "
-				+ appInfo.getAppId() + ")"
-			: // else postgres 'using' syntax:
-				"delete from schedule_entry_status a "
-				+ "using schedule_entry b "
-				+ "where b.schedule_entry_id = a.schedule_entry_id "
-				+ "and a.run_start_time < " + db.sqlDate(cutoff)
-				+ " and b.loading_application_id = " + appInfo.getAppId();
-		
-		doModify(q);
-	}
-
-	@Override
-	public void deleteScheduleStatusFor(ScheduleEntry scheduleEntry)
-		throws DbIoException
-	{
-		if (db.getDecodesDatabaseVersion() < DecodesDatabaseVersion.DECODES_DB_10)
-			return;
-		
-		String q = "delete from dacq_event where schedule_entry_status_id in "
-			+ "(select schedule_entry_status_id from schedule_entry_status where schedule_entry_id = "
-			+ scheduleEntry.getId() + ")";
-		doModify(q);
-		
-		q = "delete from platform_status where last_schedule_entry_status_id in "
-			+ "(select schedule_entry_status_id from schedule_entry_status "
-			+ "where schedule_entry_id = " + scheduleEntry.getKey() + ")";
-		doModify(q);
-		
-		q = "delete from schedule_entry_status where schedule_entry_id = "
-			+ scheduleEntry.getKey();
-		doModify(q);
-	}
-
-	@Override
-	public ScheduleEntryStatus getLastScheduleStatusFor(
-		ScheduleEntry scheduleEntry) throws DbIoException
-	{
-		if (db.getDecodesDatabaseVersion() < DecodesDatabaseVersion.DECODES_DB_10)
-			return null;
-		
-		String q = "select " + sesColumns + " from " + sesTables + " where " + sesJoinClause;
-		q = q + " and a.schedule_entry_id = " + scheduleEntry.getKey()
-			+ " and a.last_modified = "
-			+ "(select max(last_modified) from schedule_entry_status "
-			+ " where schedule_entry_id = " + scheduleEntry.getKey() + ")";
-		
-		ResultSet rs = doQuery(q);
+    }
+
+    @Override
+    public void deleteScheduleStatusFor(ScheduleEntry scheduleEntry)
+        throws DbIoException
+    {
+        if (db.getDecodesDatabaseVersion() < DecodesDatabaseVersion.DECODES_DB_10)
+		{
+            return;
+		}
+
 		try
 		{
-			while(rs != null && rs.next())
+			final StringBuilder q = new StringBuilder();
+			inTransaction(dao ->
 			{
-				ScheduleEntryStatus ses = new ScheduleEntryStatus(DbKey.createDbKey(rs, 1));
-				rs2scheduleEntryStatus(rs, ses);
-				return ses;
-			}
-			return null;
+				q.append("delete from dacq_event where schedule_entry_status_id in "
+					+ "(select schedule_entry_status_id from schedule_entry_status where schedule_entry_id = ?)");
+				doModify(q.toString(), scheduleEntry.getId());
+
+				q.setLength(0);
+				q.append("delete from platform_status where last_schedule_entry_status_id in "
+					+ "(select schedule_entry_status_id from schedule_entry_status "
+					+ "where schedule_entry_id = ?)");
+				doModify(q.toString(), scheduleEntry.getKey());
+
+				q.setLength(0);
+				q.append("delete from schedule_entry_status where schedule_entry_id = ?");
+				doModify(q.toString(), scheduleEntry.getKey());
+			});
```

Comment:
scheduleEntry.getId() and scheduleEntry.getKey() are used consistently with before changes. 

However,  If I'm reading correctly they may be the same, perhaps one should be deprecated/preferred.

Reply:
They're identical. I like getId better, but I'll add deprecated in a different PR. Pretty sure they're identical but doesn't hurt to do a scan of the code first to verify.